org.spoofax.terms/src/org/spoofax/terms/StrategoArrayListTail.java

```diff
@@ -0,0 +1,89 @@
+package org.spoofax.terms;
+
+import java.io.IOException;
+import org.spoofax.interpreter.terms.IStrategoArrayList;
+import org.spoofax.interpreter.terms.IStrategoList;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+import org.spoofax.interpreter.terms.ITermPrinter;
+import org.spoofax.terms.attachments.ITermAttachment;
+
+public class StrategoArrayListTail implements IStrategoArrayList {
```

Comment:
Why is `StrategoArrayListTail` not a `StrategoTerm`?

Reply:
`StrategoTerm` contains more fields, and extends `AbstractSimpleTerm` which contains even more fields. None of those fields are necessary or even meaningful for `StrategoArrayListTail`. This way a `StrategoArrayListTail` only contains two fields.